crates/gen/src/numerics.rs

```diff
@@ -0,0 +1,190 @@
+                "Matrix3x2" => {
+                    quote! {
+                        m11: self.m11 * rhs.m11 + self.m12 * rhs.m21,
+                        m12: self.m11 * rhs.m12 + self.m12 * rhs.m22,
+
+                        m21: self.m21 * rhs.m11 + self.m22 * rhs.m21,
+                        m22: self.m21 * rhs.m12 + self.m22 * rhs.m22,
+
+                        m31: self.m31 * rhs.m11 + self.m32 * rhs.m21 + rhs.m31,
+                        m32: self.m31 * rhs.m12 + self.m32 * rhs.m22 + rhs.m32,
+                    }
+                }
+                "Matrix4x4" => {
+                    quote! {
+                        m11: self.m11 * rhs.m11 + self.m12 * rhs.m21 + self.m13 * rhs.m31 + self.m14 * rhs.m41,
+                        m12: self.m11 * rhs.m12 + self.m12 * rhs.m22 + self.m13 * rhs.m32 + self.m14 * rhs.m42,
+                        m13: self.m11 * rhs.m13 + self.m12 * rhs.m23 + self.m13 * rhs.m33 + self.m14 * rhs.m43,
+                        m14: self.m11 * rhs.m14 + self.m12 * rhs.m24 + self.m13 * rhs.m34 + self.m14 * rhs.m44,
+
+                        m21: self.m21 * rhs.m11 + self.m22 * rhs.m21 + self.m23 * rhs.m31 + self.m24 * rhs.m41,
+                        m22: self.m21 * rhs.m12 + self.m22 * rhs.m22 + self.m23 * rhs.m32 + self.m24 * rhs.m42,
+                        m23: self.m21 * rhs.m13 + self.m22 * rhs.m23 + self.m23 * rhs.m33 + self.m24 * rhs.m43,
+                        m24: self.m21 * rhs.m14 + self.m22 * rhs.m24 + self.m23 * rhs.m34 + self.m24 * rhs.m44,
+
+                        m31: self.m31 * rhs.m11 + self.m32 * rhs.m21 + self.m33 * rhs.m31 + self.m34 * rhs.m41,
+                        m32: self.m31 * rhs.m12 + self.m32 * rhs.m22 + self.m33 * rhs.m32 + self.m34 * rhs.m42,
+                        m33: self.m31 * rhs.m13 + self.m32 * rhs.m23 + self.m33 * rhs.m33 + self.m34 * rhs.m43,
+                        m34: self.m31 * rhs.m14 + self.m32 * rhs.m24 + self.m33 * rhs.m34 + self.m34 * rhs.m44,
+
+                        m41: self.m41 * rhs.m11 + self.m42 * rhs.m21 + self.m43 * rhs.m31 + self.m44 * rhs.m41,
+                        m42: self.m41 * rhs.m12 + self.m42 * rhs.m22 + self.m43 * rhs.m32 + self.m44 * rhs.m42,
+                        m43: self.m41 * rhs.m13 + self.m42 * rhs.m23 + self.m43 * rhs.m33 + self.m44 * rhs.m43,
+                        m44: self.m41 * rhs.m14 + self.m42 * rhs.m24 + self.m43 * rhs.m34 + self.m44 * rhs.m44,
+                    }
+                }
+                _ => panic!(),
+            };
+            let impl_custom = Operation::Mul.to_tokes_with_same_custom(winrt_struct, &rows);
+            quote! {
+                #(#impl_with_same)*
+                #impl_with_scalar
+                #impl_custom
+            }
+        }
+        _ => {
+            quote! {}
+        }
+    }
+}
+
+enum Operation {
+    Add,
+    Sub,
+    Div,
+    Mul,
+}
+
+impl Operation {
+    pub fn to_tokens_with_same(&self, winrt_struct: &Struct) -> TokenStream {
```

Comment:
This is a confusing name for a method. The same what?

Reply:
In hindsight maybe it should have been named same_type or something. The method is supposed to generate all the operations against both a struct of the same type and a borrow of that type of struct.

Either way, this is being deleted in favor of just putting it all in one big quote macro.